src/System.Linq.Parallel/src/System/Linq/Parallel/QueryOperators/Inlined/NullableDecimalAverageAggregationOperator.cs

```diff
@@ -118,8 +118,11 @@ protected override bool MoveNextCore(ref Pair<decimal, long> currentElement)
 
                     if (current.HasValue)
                     {
-                        sum += current.GetValueOrDefault();
-                        count++;
+                        checked
+                        {
+                            sum += current.GetValueOrDefault();
+                            count++;
+                        }
```

Comment:
@stephentoub - ah.

I didn't think `checked` worked on decimal?  The documentation states "only on integral types".  The keyword also doesn't appear to propagate into method calls either, so this wouldn't affect that anyways (unless there's something special in the compiler?)... and the type performs its own check anyways (throws overflow regardless).

If you mean for nullable int, it lofts to long for the intermediate sum, so wont overflow before hitting the iteration limit anyways.
For nullable long it's possible, but the current behavior is non-deterministic; the worker threads used non-checked, while the final result uses checked, which means whether you get an exception or an incorrect result depends on where the partitioner split your values.  Heisenbug.

@hackcraft - this just brings it in line with the non-nullable version, which uses checked throughout.  Decimal isn't effected by checked (or unchecked), this just makes the counting safe.  I also have no way to test against a 32-bit system - my machine is 64.